ratatui-widgets/src/block.rs

```diff
@@ -1647,4 +1666,76 @@ mod tests {
         ]);
         assert_eq!(buffer, expected);
     }
+
+    fn render_merging_block_helper(buffer: &mut Buffer, merge_style: Option<&MergeStyle>) {
+        Block::bordered()
+            .merge_style(merge_style.cloned())
+            .render(Rect::new(0, 0, 3, 3), buffer);
+        Block::bordered()
+            .border_type(BorderType::Thick)
+            .merge_style(merge_style.cloned())
+            .render(Rect::new(1, 1, 3, 4), buffer);
+        Block::bordered()
+            .border_type(BorderType::Double)
+            .merge_style(merge_style.cloned())
+            .render(Rect::new(2, 3, 3, 3), buffer);
+        Block::bordered()
+            .border_type(BorderType::Rounded)
+            .merge_style(merge_style.cloned())
+            .render(Rect::new(3, 0, 3, 2), buffer);
+        Block::bordered()
+            .merge_style(merge_style.cloned())
+            .render(buffer.area, buffer);
+    }
+
+    #[test]
+    fn render_non_merging_blocks() {
```

Comment:
Are these t3 ests sufficient to cover all the cases? Maybe use `rstest` and add some different test variants to each one.